aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,1105 @@
+                voting_duration,
+                min_required_proposer_voting_power,
+                next_proposal_id: 0,
+                dao_signer_capability: res_cap,
+                admin: src_addr,
+                pending_admin: option::none(),
+            },
+        );
+        move_to(
+            &res_signer,
+            Proposals {
+                proposals: table::new()
+            }
+        );
+
+        move_to(
+          &res_signer,
+            ProposalVotingStatistics {
+                proposals: table::new()
+            }
+        );
+
+        let dao_addr = signer::address_of(&res_signer);
+
+        emit_create_dao_event(
+            &res_signer,
+            name,
+            threshold,
+            voting_duration,
+            min_required_proposer_voting_power,
+            voting_token_collection_creator,
+            collection_name,
+            src_addr,
+        );
+        dao_addr
+    }
+
+    /// Only DAO Goverance token holders can create proposal
+    public entry fun create_proposal(
+        account: &signer,
+        nft_dao: address, // resource account address of the nft dao
+        name: String, // name of the proposal
+        description: String, // description of the proposal
+        function_name: String, // only two functions: (1) "no_op" and (2) "transfer_fund". Only transfer_fund has correponding function now.
+        arg_names: vector<String>, // name of the arguments of the function to be called. The arg here should be the same as the argument used in the function
+        arg_values: vector<vector<u8>>, // bcs serailized values of argument values
+        arg_types:vector<String>, // types of arguments. currently, we only support string, u8, u64, u128, bool, address.
+        start_time_sec: u64, // when the voting starts
+        token_names: vector<String>, // The name of the token, the proposer want to use for proposing
+        property_versions: vector<u64>, // the property versions of the corresponding tokens, the proposer want to use for proposing
+    ) acquires DAO, Proposals {
+        let dao = borrow_global_mut<DAO>(nft_dao);
+        assert!(string::length(&name) <= 64, error::invalid_argument(ESTRING_TOO_LONG));
+        assert!(string::length(&description) <= 512, error::invalid_argument(ESTRING_TOO_LONG));
+
+        // verify the account's token has enough weights to create proposal
+        let weights = get_proposal_weights(account, &token_names, &property_versions, dao);
+        assert!(weights >= dao.min_required_proposer_voting_power, error::permission_denied(EVOTING_POWER_NOT_ENOUGH));
+
+        // verify the parameters are legit
+        let pm = property_map::new(arg_names, arg_values, arg_types);
+        assert_function_valid(function_name, &pm);
+
+        // verify the start_time is in future
+        let now = timestamp::now_seconds();
+        assert!(start_time_sec > now, error::invalid_argument(EPROPOSAL_START_TIME_SHOULD_BE_IN_FUTURE));
```

Comment:
Didn't catch this before - Why do we want to allow users to set start_time_sec? Why not just take the current timestamp?

Reply:
Discussed offline. This is just for the proposer to specify a time range in the future that they can reach to the voters and get them prepared for voting.